### PR TITLE
Add custom:region attribute to Cognito User Pool and OSCER User provisioning

### PR DESCRIPTION
**Pull request by nkutub**
## Summary

Adds `custom:region` attribute to Cognito User Pool schema to support region-based access control for staff users.

**Changes:**
- Added `region` custom attribute to Cognito User Pool schema (accessed as `custom:region` in tokens)
- Added `custom:region` to client's `read_attributes` and `write_attributes`
- Added `custom:region` to the user provisioning logic to sync user region with IdP attribute

## Why

The application uses region-based task assignment and access control. Staff users need a `region` attribute stored in Cognito so:
1. Region is included in ID tokens after authentication
2. Application can sync region to user records during SSO login
3. Policy scopes can filter tasks/cases by user's region

## Preview environment for reporting-app
♻️ Environment destroyed ♻️

### CODE REVIEW

File: reporting-app/spec/services/staff_user_provisioner_spec.rb [hunk 1]
```diff
@@ -109,6 +125,16 @@
           .from("caseworker")
           .to("admin")
       end
+
+      it "updates region when changed in IdP" do
```
Comment (baonguyenNava) [inline reporting-app/spec/services/staff_user_provisioner_spec.rb]: I think we need a test to change an existing user's region to nil.

**Review by baonguyenNava — APPROVED** [review]
